File: core/src/main/java/dev/bee/kanjianki/core/ReminderNotificationPolicy.java
package dev.bee.kanjianki.core;

public final class ReminderNotificationPolicy {
    private ReminderNotificationPolicy() {
    }

    public static boolean notificationsAllowed(
            boolean hasRuntimeNotificationPermission,
            boolean notificationsEnabled,
            boolean channelBlocked
    ) {
        return hasRuntimeNotificationPermission && notificationsEnabled && !channelBlocked;
    }
}

File: core/src/main/kotlin/dev/bee/kanjianki/core/ReminderNotificationPolicy.kt
package dev.bee.kanjianki.core

object ReminderNotificationPolicy {
    @JvmStatic
    fun notificationsAllowed(
        hasRuntimeNotificationPermission: Boolean,
        notificationsEnabled: Boolean,
        channelBlocked: Boolean,
    ): Boolean {
        return hasRuntimeNotificationPermission && notificationsEnabled && !channelBlocked
    }
}
